src/Aspire.Cli/Commands/UpdateCommand.cs

```diff
@@ -233,15 +246,27 @@ private async Task<int> ExecuteSelfUpdateAsync(ParseResult parseResult, Cancella
     {
         var channel = selectedChannel ?? parseResult.GetValue<string?>("--channel") ?? parseResult.GetValue<string?>("--quality");
 
-        // If channel is not specified, prompt the user
+        // If channel is not specified, check for hives and either prompt or use default
         if (string.IsNullOrEmpty(channel))
         {
-            var channels = new[] { "stable", "staging", "daily" };
-            channel = await InteractionService.PromptForSelectionAsync(
-                "Select the channel to update to:",
-                channels,
-                q => q,
-                cancellationToken);
+            // If there are hives (PR build directories), prompt for channel selection.
+            // Otherwise, use "stable" as the default channel.
+            var hasHives = ExecutionContext.HasHives;
+            
+            if (hasHives)
+            {
+                var channels = new[] { "stable", "staging", "daily" };
+                channel = await InteractionService.PromptForSelectionAsync(
+                    "Select the channel to update to:",
+                    channels,
+                    q => q,
+                    cancellationToken);
+            }
+            else
+            {
+                // Use stable as the default channel for self-update
+                channel = "stable";
```

Comment:
Can you move stable, staging, daily to constants somewhere and reuse everywhere? I've noticed the range and names of channels tends to change over time